Windows Clusters Page Refresh - RKE2 Instructions (#1306)
rancher/rancher-docs: docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/use-windows-clusters/use-windows-clusters.md

  <link rel="canonical" href="https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/use-windows-clusters"/>
</head>

> When provisioning a [custom cluster](../../../../reference-guides/cluster-configuration/rancher-server-configuration/use-existing-nodes/use-existing-nodes.md) using Rancher, RKE2 is used to install Kubernetes on your existing nodes.

Author: Updated to active voice

> When provisioning a [custom cluster](../../../../reference-guides/cluster-configuration/rancher-server-configuration/use-existing-nodes/use-existing-nodes.md) using Rancher, Rancher uses RKE2, also known as RKE Government, to install Kubernetes on your existing nodes.

Reviewer: As Billy noted, we don't use "RKE Government" too often: https://github.com/rancher/rancher-docs/pull/1305#discussion_r1616430202

Author: Removed the reference across versions

In a Windows cluster provisioned with Rancher, the cluster must contain both Linux and Windows nodes. The Kubernetes controlplane can only run on Linux nodes, and the Windows nodes can only have the worker role. Windows nodes can only be used for deploying workloads.

> Other requirements for Windows clusters include:

Author: Updated and removed this section as the material is repeated in the main requirement section.

Some other requirements for Windows clusters include:
Other requirements for Windows clusters include:

- You can only add Windows nodes to a cluster if Windows support is enabled when the cluster is created. Windows support cannot be enabled for existing clusters.

> - Kubernetes 1.15+ is required.

Author: Removed K8s 1.15 reference and added RKE2 support matrices link.

> 

Author: Updated the header and intro sentence

- HostProcess containers in Windows RKE2 are supported in Kubernetes v1.24.1 and up. See [the upstream documentation](https://kubernetes.io/docs/tasks/configure-pod-container/create-hostprocess-pod/) for more information.

## General Requirements

> The general node requirements for networking and operating systems are the same as the node requirements for a [Rancher installation](../../../../getting-started/installation-and-upgrade/installation-requirements/installation-requirements.md).

Reviewer: ```suggestion
The general networking and operating system requirements for Windows nodes are the same as for other [Rancher installations](../../../../getting-started/installation-and-upgrade/installation-requirements/installation-requirements.md).
```

### OS Requirements

Our support for Windows Server and Windows containers match the Microsoft official lifecycle for LTSC (Long-Term Servicing Channel) and SAC (Semi-Annual Channel).

For the support lifecycle dates for Windows Server, see the [Microsoft Documentation.](https://docs.microsoft.com/en-us/windows-server/get-started/windows-server-release-info)

### Kubernetes Version

> Information regarding Kubernetes component versions can be viewed in the [support matrices for RKE2 versions](https://www.suse.com/suse-rke2/support-matrix/all-supported-versions/).

Reviewer: ```suggestion
For more information regarding Kubernetes component versions, see the [support matrices for RKE2 versions](https://www.suse.com/suse-rke2/support-matrix/all-supported-versions/).
```

## Requirements for Windows Clusters

The general node requirements for networking and operating systems are the same as the node requirements for a [Rancher installation](../../../../getting-started/installation-and-upgrade/installation-requirements/installation-requirements.md).

### OS Requirements

Our support for Windows Server and Windows containers match the Microsoft official lifecycle for LTSC (Long-Term Servicing Channel) and SAC (Semi-Annual Channel).

> 

Author: [Verified with support](https://github.com/rancher/rancher-docs/issues/1074#issuecomment-2146242652) and added the support matrices link as a k8s component versioning reference in the `Kubernetes Version` section.

Before provisioning a new cluster, be sure that you have already installed Rancher on a device that accepts inbound network traffic. This is required in order for the cluster nodes to communicate with Rancher. If you have not already installed Rancher, please refer to the [installation documentation](../../../../getting-started/installation-and-upgrade/installation-and-upgrade.md) before proceeding with this guide.

Rancher supports Windows using Calico or Flannel as the network provider.

> There are two network options: [**Host Gateway (L2bridge)**](https://github.com/coreos/flannel/blob/master/Documentation/backends.md#host-gw) and [**VXLAN (Overlay)**](https://github.com/coreos/flannel/blob/master/Documentation/backends.md#vxlan). The default option is **VXLAN (Overlay)** mode.

Reviewer 2: Since [line 188](https://github.com/rancher/rancher-docs/pull/1306/files#diff-1166c01078611b4717dd6490197b417b7ea01caa942801e80e93c1219c55d9b3L188) is getting dropped, are these descriptions about the two network options still relevant?

Author: Verified this with support as well and they are not relevant moving forward, went ahead and removed them. Thanks for the review Billy!

This tutorial describes how to create a Rancher-provisioned cluster with the three nodes in the [recommended architecture.](#recommended-architecture)

> When you provision a cluster with Rancher on existing nodes, you will add nodes to the cluster by installing the [Rancher agent](../../../../reference-guides/cluster-configuration/rancher-server-configuration/use-existing-nodes/rancher-agent-options.md) on each one. When you create or edit your cluster from the Rancher UI, run the **Registration Command** on each server to add it to your cluster.

Reviewer: ```suggestion
When you provision a cluster with Rancher on existing nodes, you add nodes to the cluster by installing the [Rancher agent](../../../../reference-guides/cluster-configuration/rancher-server-configuration/use-existing-nodes/rancher-agent-options.md) on each one. To create or edit your cluster from the Rancher UI, run the **Registration Command** on each server to add it to your cluster.
```

This tutorial describes how to create a Rancher-provisioned cluster with the three nodes in the [recommended architecture.](#recommended-architecture)

> When you provision a cluster with Rancher on existing nodes, you will add nodes to the cluster by installing the [Rancher agent](../../../../reference-guides/cluster-configuration/rancher-server-configuration/use-existing-nodes/rancher-agent-options.md) on each one. When you create or edit your cluster from the Rancher UI, you will see a **Registration Command** that you can run on each server to add it to your cluster.

Reviewer: ```suggestion
When you provision a cluster with Rancher on existing nodes, you will add nodes to the cluster by installing the [Rancher agent](../../../../reference-guides/cluster-configuration/rancher-server-configuration/use-existing-nodes/rancher-agent-options.md) on each one. When you create or edit your cluster from the Rancher UI, run the **Registration Command** on each server to add it to your cluster.
```

Author: Updated with suggestion

- `Default`, containing the `default` namespace
- `System`, containing the `cattle-system`, `ingress-nginx`, `kube-public`, and `kube-system` namespaces

It may take a few minutes for the node to be registered in your cluster.

#### Add Linux Worker Node

In this section, we run a command to register the Linux worker node to the cluster.

> After the initial provisioning of your cluster, your cluster only has a single Linux host. Next, we add another Linux `worker` host, which will be used to support the _Rancher cluster agent_, _Metrics server_, _DNS_ and _Ingress_ for your cluster.

Reviewer: ```suggestion
After the initial provisioning of your cluster, your cluster only has a single Linux host. Add another Linux `worker` host to support the _Rancher cluster agent_, _Metrics server_, _DNS_ and _Ingress_ for your cluster.
```

Author: Updated with suggestion

In this section, we run a command to register the Windows worker node to the cluster.

:::note

> The cluster must be up and running with Linux etcd, control plane, and worker nodes before the registration command for adding Windows workers will display.

Reviewer: ```suggestion
The registration command to add the Windows workers only appears after the cluster is running with Linux etcd, control plane, and worker nodes.
```
